src/zenml/utils/source_utils.py

```diff
@@ -238,32 +241,61 @@ def set_custom_local_repository(
 ) -> None:
     """Manually defines a local repository for a path.
 
+    To explain what this function does we need to take a dive into source
```

Comment:
😆

Reply:
this must be a new record for longest legit docstring in ZenML (aside from CLI init etc(

Reply:
I think @stefannica has written longer and better ones than this :P